echopype/mask/api.py

```diff
@@ -0,0 +1,306 @@
+        raise RuntimeError("Either freqAB or chanAB must be given!")
+    elif (freqAB is not None) and (chanAB is not None):
+        raise RuntimeError("Only freqAB or chanAB must be given, but not both!")
+    elif freqAB is not None:
+        if not isinstance(freqAB, list):
+            raise TypeError("freqAB must be a list!")
+        elif len(set(freqAB)) != 2:
+            raise RuntimeError("freqAB must be a list of length 2 with unique elements!")
+    else:
+        if not isinstance(chanAB, list):
+            raise TypeError("chanAB must be a list!")
+        elif len(set(chanAB)) != 2:
+            raise RuntimeError("chanAB must be a list of length 2 with unique elements!")
+
+    # check that operator is a string and a valid operator
+    if not isinstance(operator, str):
+        raise TypeError("operator must be a string!")
+    else:
+        if operator not in [">", "<", "<=", ">=", "=="]:
+            raise RuntimeError("Invalid operator!")
+
+    # ensure that diff is a float or an int
+    if not isinstance(diff, (float, int)):
+        raise TypeError("diff must be a float or int!")
+
+
+def _check_source_Sv_freq_diff(
+    source_Sv: Union[xr.Dataset, str, pathlib.Path],
+    storage_options: Optional[dict],
+    freqAB: Optional[List[float]] = None,
+    chanAB: Optional[List[str]] = None,
+) -> Union[xr.Dataset, xr.DataArray]:
+    """
+    Ensures that ``source_Sv`` is of the correct type, it exists if it is a
+    path, contains ``channel`` as a coordinate and ``frequency_nominal`` as a
+    variable, and that the provided list input (``freqAB`` or ``chanAB``) are
+    contained in the coordinate ``channel`` or variable ``frequency_nominal``.
+
+    Parameters
+    ----------
+    source_Sv: xr.Dataset or str or pathlib.Path
+        If a Dataset this value contains the Sv data to create a
+        mask for, else it specifies the path to a zarr or netcdf file
+    storage_options: dict, optional
+        Any additional parameters for the storage backend, corresponding to the
+        path provided for ``source_Sv``
+    freqAB: list of float, optional
+        The pair of nominal frequencies to be used for frequency-differencing, where
+        the first element corresponds to ``freqA`` and the second element corresponds
+        to ``freqB``
+    chanAB: list of float, optional
+        The pair of channels that will be used to select the nominal frequencies to be
+        used for frequency-differencing, where the first element corresponds to ``freqA``
+        and the second element corresponds to ``freqB``
+
+    Returns
+    -------
+    source_Sv: xr.Dataset
+        A Dataset containing the Sv data
+    """
+
+    source_Sv, file_type = validate_source_Sv(source_Sv, storage_options)
+
+    if isinstance(source_Sv, str):
+
+        # open up Dataset using source_Sv path
+        source_Sv = xr.open_dataset(source_Sv, engine=file_type, chunks="auto", **storage_options)
```

Comment:
I think that these few lines should be moved outside of this function and into the main block of `frequency_difference`, and that way you can have this function `_check_source_Sv_freq_diff` really just really handles the checks, instead of returning a Dataset that is somewhat awkward and incompatible with the function name.